mdms/mdms-analytics/src/main/scala/de/hpi/isg/mdms/analytics/TableImportance.scala

```diff
@@ -0,0 +1,144 @@
+
+    // qEntT ... product of q and entropy summed over the entire table
+    val qEntT = qEnt
+      .map(cs => (idUtils.getTableId(cs._1), cs._2))
+      .reduceByKey(_._1, (a, b) => (a._1, a._2 + b._2))
+
+    // R ... number of tuples of table
+    val R = metadataStore.loadConstraints(tupleCount)
+      .map(tp => (tp.getTableId, tp.getNumTuples))
+
+    // logRqEntT ... sum of log(numTuples) and qEntT
+    val logRqEntT = qEntT
+      .keyBy(_._1).keyJoin(R.keyBy(_._1)).assemble((a, b) => (a._1, a._2 + math.log(b._2)))
+
+    // edgeCol ... Start and target of edge (COLUMN-ID used here)
+    val edgeCol = metadataStore.loadConstraints(inclusionDependency)
+      .map(ind => (ind.getDependentColumnIds.apply(0), ind.getReferencedColumnIds.apply(0)))
+
+    // Entropy of each column
+    val entCol = metadataStore.loadConstraints(columnStatistics)
+      .map(ind => (ind.getColumnId, ind.getEntropy))
+
+    val entEdgeT = edgeCol
+      .keyBy(_._1).keyJoin(entCol.keyBy(_._1)).assemble((a, b) => (a._1, a._2, b._2))
+      .map(a => (idUtils.getTableId(a._1), idUtils.getTableId(a._2), a._3))
+
+    // probT ... probability of each table
+    val probT = entEdgeT
+      .keyBy(_._1).keyJoin(logRqEntT.keyBy(_._1)).assemble((a, b) => (a._1, a._2, a._3 / b._2))
+
+    // Calculate the probability P[R,S]
+    val probabilityRS = probT
+      .filter(a => a._1 != a._2).reduceByKey(t => (t._1, t._2), (a, b) => (a._1, a._2, a._3 + b._3))
+
+    // Getting dummy values for P[R,R]
+    val dummyProbabilityRR = metadataStore.loadTables().map(tableMock => (tableMock.id, tableMock.id, 0.0))
+
+    // Union between P[R,S] + (dummy) P[R,R]
+    val dummyProbabilityRSRR = probabilityRS.union(dummyProbabilityRR)
+
+    // Calculate final P[R,R]
+    val probabilityRR = dummyProbabilityRSRR.reduceByKey(_._1, (a, b) => (a._1, a._2, a._3 + b._3))
+      .map(a => (a._1, a._1, 1.0 - a._3))
+
+    // Union between P[R,S] + P[R,R]
+    val probability = probabilityRR.union(probabilityRS)
+
+    // Return probability matrix
+    return probability
+  }
+
+  /**
+    * Finding the table importance
+    *
+    * @param probMatrix stores the probability matrix
+    * @param tupleCount is the constraint collection tuple count
+    * @param metadataStore  is the metadata store
+    * @param idUtils utility tools
+    * @param columnStatistics is constraint collection column statistics
+    * @param epsilon if Euclidean Distance between V and V+1  is
+    *                lower epsilon, the stationary distribution is reached
+    * @param numIteration is the maximum number of iteration
+    * @param maxnumOrEpsilon true: solution is found if maximum number of iteration is reached
+    *                        false: solution is found if changes of solution vector is below a critical threshold
+    * @return the solution vector containing the table importance
+    */
+  def tableImport(probMatrix: DataQuanta[(Int, Int, Double)],
```

Comment:
It would be great if we overload that method, such that we have a version that already calculates the probability matrix.